webpush/push.go

```diff
@@ -69,6 +69,25 @@ func NewPushRequest(sub *Subscription, message string, token string) (*http.Requ
 	return req, nil
 }
 
+// NewVapidRequest creates a valid Web Push HTTP request for sending a message
+// to a subscriber, using Vapid authentication. You can add more headers to
+// configure collapsing, TTL.
+func NewVapidRequest(sub *Subscription, message string, vapid *Vapid) (*http.Request, error) {
```

Comment:
@gauntface merely Go idiomatic naming convention: https://github.com/golang/go/wiki/CodeReviewComments#initialisms
But the other issue remains: if I haven't seen the package before, I probably wouldn't know whether I should use `NewPushRequest` or the `NewVAPRequest` and when.

I don't know what the usage is like of this package, but maybe there's still time to just make a breaking change now, have one function which works for both, with and without VAP.